### PR TITLE
[LiveMetricsExporter] Add Filtering Support part 1 - Take in filter parameters

### PR DESCRIPTION
**Pull request by xiang17**
This is the first of a series of PRs for Adding Filtering support for the Live Metrics feature. More PRs are to follow.

The series is roughly broken down into these parts:
* [This PR] Take CollectionConfigurationInfo and store it. It's part of the Breeze response used for storing the Query (Filtering and Projection parameters etc.)
* Parse the Query parameters and store the compiled Filters and Projections.
* Apply the Filters and Projections.
* Aggregates the result and add them to the Post request to be sent.
* Handle failure scenario: parsing "x-ms-qps-subscribed" header failed.
* Handle failure scenario: sending data failed while in Post state.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by jsquire — DISMISSED** [review]
@xiang17 :

Your account needs adjustments before this PR can move forward.  As an internal contributor, you'll need to have your Microsoft and Azure organization memberships public and request write access to the repository.

Please review the Azure SDK onboarding documentation and use the associated Teams channel for support/discussion.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs:113]:
```suggestion
                    if (!response.Subscribed)
```

Responding to your comment on line 118.
FYI, line 113 should say `if (!response.Subscribed)`.
That's a copy-paste bug I introduced. I fixed this in my other PR that hasn't merged yet. 
You can go ahead and make the bug fix here, and I think you can delete your comment on 118.
Sorry for the interruption.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Customizations/QuickPulseSDKClientAPIsRestClient.cs
Comment (xiang17) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Customizations/QuickPulseSDKClientAPIsRestClient.cs:53]: Good catch! I've updated them.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs]: Update, I'm fixing this in #41545.

I think you can delete your comment on line 118. Please correct me if I'm misunderstanding this comment.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs:124]: I think you can remove the `etagUpdated` bool if you move the `response.CollectionConfigurationInfo != null` check into the if block at line 106.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs]: The other PR just merged. It's showing as a merge conflict, but if you merge `main` into your branch it should auto resolve.
Again, sorry for the interruption.

**Comment by xiang17** [comment]
> @xiang17 :
> 
> Your account needs adjustments before this PR can move forward. As an internal contributor, you'll need to have your Microsoft and Azure organization memberships public and request write access to the repository.
> 
> Please review the Azure SDK onboarding documentation and use the associated Teams channel for support/discussion.
> 
> * [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
> * [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_
> 
> Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

Thanks for your guidance. I have requested and have been granted write access to the repository.

Can you check my status now and see if you can unblock the PR?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (xiang17) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs:124]: If I move it there, the `_collectionConfigurationInfo` might be updated without checking whether `response.Subscribed` is true. Even though server shouldn't send us contradictory data like this.

I've rearranged the check order a bit: we check `Subscribed` value first, then does the `response.CollectionConfigurationInfo` check inside the `else` block.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.RestClient.cs:122]: nit:
```suggestion
        private void RefreshConfigurationOnEtagChange(QuickPulseResponse response)
```

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by TimothyMothra — APPROVED** [review]
LGTM!

**Pull request merged**
